jobs/build-webapp.groovy

```diff
@@ -46,6 +46,11 @@ new Setup(steps
 // We only need to lock out for promoting; builds are fine to do in parallel.
 ).allowConcurrentBuilds(
 
+// Sometimes we want to debug a deploy a week or two later.  Let's
+// keep a lot of these.
+).resetNumBuildsToKeep(
+   500,
```

Comment:
This should give us a good window of logs we keep around, which is great.  I wonder if we should copy logs for failed build/tests/deploy somewhere where we dont delete them.  That should hopefully help us keep the likely more interesting logs longer since its not as many of them.

Reply:
Ooh, interesting idea!  I don't know how to move logs in ways that are useful to view them still.

Another approach would be to mark all failed builds as "keep forever", and then have a cronjob that unkeeps old-enough failed jobs after a while.  But then if folks manually mark a build as "keep forever" then that won't be respected.

Reply:
oh!  that's actually a really idea too! and probably easier than saving files somewhere...  Cool - I created this ticket https://khanacademy.atlassian.net/browse/INFRA-9507.  I think it's worth while doing.